src/main/java/org/elasticsearch/index/gateway/IndexShardGateway.java

```diff
@@ -184,36 +184,20 @@ public void recover(boolean indexShouldExists, RecoveryState recoveryState) thro
                 return;
             }
 
-            // move an existing translog, if exists, to "recovering" state, and start reading from it
-            Translog translog = indexShard.translog();
+            final Translog translog = indexShard.translog();
             final Path translogName = translog.getPath(translogId);
-            final Path recoverTranslogName = translogName.resolveSibling(translogName.getFileName() + ".recovering");
-
             logger.trace("try recover from translog file {} locations: {}", translogName, Arrays.toString(translog.locations()));
             Path recoveringTranslogFile = null;
+            OUTER:
             for (Path translogLocation : translog.locations()) {
-                final Path tmpRecoveringFile = translogLocation.resolve(recoverTranslogName);
-                if (Files.exists(tmpRecoveringFile) == false) {
-                    Path tmpTranslogFile = translogLocation.resolve(translogName);
-                    if (Files.exists(tmpTranslogFile)) {
-                        logger.trace("Translog file found in {} - renaming", translogLocation);
-                        for (int i = 0; i < RECOVERY_TRANSLOG_RENAME_RETRIES; i++) {
-                            try {
-                                Files.move(tmpTranslogFile, tmpRecoveringFile, StandardCopyOption.ATOMIC_MOVE);
-                                recoveringTranslogFile = tmpRecoveringFile;
-                                logger.trace("Renamed translog from {} to {}", tmpTranslogFile.getFileName(), recoveringTranslogFile.getFileName());
-                                break;
-                            } catch (Exception ex) {
-                                logger.debug("Failed to rename tmp recovery file", ex);
-                            }
-                        }
-                    } else {
-                        logger.trace("Translog file NOT found in {} - continue", translogLocation);
-                    }
-                } else {
-                    recoveringTranslogFile = tmpRecoveringFile;
-                    break;
+                // we have to support .recovering since it's a leftover from previous version but might still be on the filesystem
+                // we used to rename the foo into foo.recovering since foo was reused / overwritten but we fixed that in 1.5
```

Comment:
double checking before I change it - the comment "overwritten but we fixed that in 1.5" is wrong right? it was fixed in 2.0 as far as I can tell.